### PR TITLE
Userlist bugs

### PR DESCRIPTION
**Pull request by reidbaker**
This fixes the userlist stuff in the menu fragment.

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java [hunk 1]
```diff
@@ -105,8 +108,11 @@ public void onClick(View v) {
             }
         });
         
-        ListView userView = (ListView)v.findViewById(R.id.connected_users);
-        userView.setAdapter(this.userAdapter);
+        //ListView userView = (ListView)v.findViewById(R.id.connected_users);
```
Comment (bryansills) [inline SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java]: remove commented code

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java [hunk 2]
```diff
@@ -122,8 +125,9 @@ public void onClick(View v) {
         disconnect = (Button)v.findViewById(R.id.disconnect_btn);
         disband = (Button)v.findViewById(R.id.disband_btn);
 
-        ListView users = (ListView) v.findViewById(R.id.connected_users);
-        users.setAdapter(this.adapter);
+        userView = (LinearLayout) v.findViewById(R.id.connected_users);
+        updateUserView();
+        //users.setAdapter(this.adapter);
```
Comment (bryansills) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:130]: more commented code

**Comment by bryansills** [comment]
other than the commented code looks good. is there an established format for tabbing with xml so that we don't have pull requests that look like huge changes, but only change the tabbing?

File: SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java [hunk 3]
```diff
@@ -178,4 +185,14 @@ private UserListService getUserListService(){
         }
         return userService;
     }
+    
+    private void updateUserView(){
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/MenuFragment.java:189]: This feels wrong...given the circumstances, I'm not going to hold up the pull request but I'd like to know why we have to do this vs using a listview.

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java [hunk 4]
```diff
@@ -122,8 +125,9 @@ public void onClick(View v) {
         disconnect = (Button)v.findViewById(R.id.disconnect_btn);
         disband = (Button)v.findViewById(R.id.disband_btn);
 
-        ListView users = (ListView) v.findViewById(R.id.connected_users);
-        users.setAdapter(this.adapter);
+        userView = (LinearLayout) v.findViewById(R.id.connected_users);
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:128]: Dont like this...see above comment.

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java [hunk 5]
```diff
@@ -394,4 +400,14 @@ private UserListService getUserListService(){
         }
         return userService;
     }
+    
+    private void updateUserView(){
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:404]: Duplicate code (MenuFragment), and I repeat my objection.

Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:404]: We do this in multiple places - In fact, anywhere that has the userlist service has two methods that are duplicated in every fragment that needs to get at the service and it's contents. I suppose we could work around it by making this a static method somewhere

**Comment by theJenix** [comment]
After talking to @ejohnson44 my objections are retracted...somewhat.  I'm going to pull this in, but since we use the same strategy in both NetworkFragment and MenuFragment, please add a card create a reusable UserListView component.

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java (hunk 5 above)
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:404]: If we're going to stay with this strategy, we should define a UserListView that either extends View or LinearLayout, and exposes this functionality.  That way we can just reuse the component.

Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:404]: That was my plan if I can't find another way to do this, but I didn't have the time to work with that to get this done for the demo today. I made a card in trello, so I will get to it soonish

**Pull request merged**